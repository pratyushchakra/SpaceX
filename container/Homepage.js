import Portfolio from '../components/Portfolio/Portfolio';
import styles from './Homepage.style';
// import { Helment } from 'react-helmet';
const Homepage = () => (
    <div>
        {/* <Helment>
            <title>SpaceX App</title>
            <meta property="og:title" content="SpaceX App"></meta>
        </Helment> */}
        <header>
            <h1>SpaceX Launch Program</h1>
        </header>
        <section>
            <Portfolio />
        </section>
        <footer>
            <b>Developed By:</b><p>Pratyush Chakra</p>
        </footer>
        <style jsx>{styles}</style>
    </div>
);

export default Homepage