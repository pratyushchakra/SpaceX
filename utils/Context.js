import React, { useState, useContext, useEffect } from 'react';
import createContainer from 'constate';

const nameHOC = (Component, suffix = '') => {
  return `${Component.originalName ||
    Component.displayName ||
    Component.name ||
    'Component'}${suffix}`;
};

const formatData = (allLaunchesData = []) => {
  return allLaunchesData.map(v => {
    return {
      ...v,
      isLandingTrue: v.rocket.first_stage.cores[v.rocket.first_stage.cores.length - 1].land_success
    }
  })
}

function useUserContext({ data: initialData }) {
  const [allLaunchesData, setLaunchData] = useState(initialData);
  const [filterData, setFilterData] = useState({
    launch: null, landing: null, year: ''
  });
  return {
    launchConfig: {
      allLaunches: [
        ...formatData(allLaunchesData)
      ],
      setLaunchData
    },
    filterConfig: {
      filterData,
      setFilterData
    }
  }
}

const LaunchInfoContext = createContainer(useUserContext);

const withContext = (WrappedComponent) => {
  const ComponentWithData = (props) => {
    const store = useContext(LaunchInfoContext.Context);
    return <WrappedComponent {...props} store={store} />;
  };

  ComponentWithData.originalName = nameHOC(WrappedComponent);
  ComponentWithData.displayName = nameHOC(
    WrappedComponent
  );
  return ComponentWithData;
};

export default LaunchInfoContext;
export { withContext };
