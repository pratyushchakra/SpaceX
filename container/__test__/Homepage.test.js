import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Context from '../../utils/Context'
import Homepage from './../Homepage';
import mockData from '../../utils/mock'
Enzyme.configure({ adapter: new Adapter() })

describe('Homepage', () => {
    it('should render as expected', () => {
        const wrapper = shallow(
            <Homepage />
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('should render Portfolio component', () => {
        const wrapper = shallow(
            <Homepage />
        );
        expect(wrapper.find('Portfolio')).toBeTruthy()
    });
    it('renders the component deeply', () => {
        const wrapper = mount(
            <Context.Provider data={mockData}>
                <Homepage />
            </Context.Provider>
        );
        expect(wrapper.find('Pod').length).toBe(2);
        expect(wrapper.find('Filter')).toBeTruthy()
    })
});