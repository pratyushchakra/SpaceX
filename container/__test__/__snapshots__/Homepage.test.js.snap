// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render as expected 1`] = `
"<div className=\\"jsx-2218827264\\">
  <header className=\\"jsx-2218827264\\">
    <h1 className=\\"jsx-2218827264\\">
      SpaceX Launch Program
    </h1>
  </header>
  <section className=\\"jsx-2218827264\\">
    <Portfolio />
  </section>
  <footer className=\\"jsx-2218827264\\">
    <b className=\\"jsx-2218827264\\">
      Developed By:
    </b>
    <p className=\\"jsx-2218827264\\">
      Pratyush Chakra
    </p>
  </footer>
  <JSXStyle id=\\"2218827264\\">
    h
    t
    m
    l
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    e
    e
    e
    ;
    }
    p
    ,
    b
    {
    m
    a
    r
    g
    i
    n
    :
    0
    ;
    }
    b
    u
    t
    t
    o
    n
    {
    o
    u
    t
    l
    i
    n
    e
    :
    0
    ;
    b
    o
    r
    d
    e
    r
    :
    0
    ;
    c
    u
    r
    s
    o
    r
    :
    p
    o
    i
    n
    t
    e
    r
    ;
    }
    f
    o
    o
    t
    e
    r
    {
    w
    i
    d
    t
    h
    :
    -
    w
    e
    b
    k
    i
    t
    -
    m
    a
    x
    -
    c
    o
    n
    t
    e
    n
    t
    ;
    w
    i
    d
    t
    h
    :
    -
    m
    o
    z
    -
    m
    a
    x
    -
    c
    o
    n
    t
    e
    n
    t
    ;
    w
    i
    d
    t
    h
    :
    m
    a
    x
    -
    c
    o
    n
    t
    e
    n
    t
    ;
    m
    a
    r
    g
    i
    n
    :
    0
     
    a
    u
    t
    o
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    }
  </JSXStyle>
</div>"
`;
